docs/articles_en/documentation/openvino-extensibility/transformation-api/model-pass.rst

```diff
@@ -11,6 +11,13 @@ OpenVINO Model Pass
 
 ``ov::pass::ModelPass`` is used for transformations that take entire ``ov::Model`` as an input and process it.
```

Comment:
I think we need to add a few examples of use cases here.
And maybe some recommendations/warnings. 
E.g. Please use ModelPass only if Mather pass  (add link here) is not fit your needs. Be careful with Parameter/Result modification, especially deleting or adding new ones. It may break the original model and affect users.   
Etc